pkg/process/events/listener_linux_test.go

```diff
@@ -0,0 +1,188 @@
+package events
+
+import (
+	"context"
+	"errors"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/pkg/process/events/model"
+	"github.com/DataDog/datadog-agent/pkg/security/api"
+	"github.com/DataDog/datadog-agent/pkg/security/api/mocks"
+)
+
+// assertProcessEvents compares two ProcessEvents. Two events can't be compared using directly assert.Equal
+// due to the embedded time fields
+func assertProcessEvents(t *testing.T, expected, actual *model.ProcessEvent) {
+	assert.Equal(t, expected.EventType, actual.EventType)
+	assert.WithinDuration(t, expected.CollectionTime, actual.CollectionTime, 0)
+	assert.Equal(t, expected.Pid, actual.Pid)
+	assert.Equal(t, expected.Ppid, actual.Ppid)
+	assert.Equal(t, expected.UID, actual.UID)
+	assert.Equal(t, expected.GID, actual.GID)
+	assert.Equal(t, expected.Username, actual.Username)
+	assert.Equal(t, expected.Group, actual.Group)
+	assert.Equal(t, expected.Exe, actual.Exe)
+	assert.Equal(t, expected.Cmdline, actual.Cmdline)
+	assert.WithinDuration(t, expected.ForkTime, actual.ForkTime, 0)
+	assert.WithinDuration(t, expected.ExecTime, actual.ExecTime, 0)
+	assert.WithinDuration(t, expected.ExitTime, actual.ExitTime, 0)
+}
+
+// TestProcessEventFiltering asserts that sysProbeListener collects only expected events and drops everything else
+func TestProcessEventFiltering(t *testing.T) {
+	rawEvents := make([]*model.ProcessMonitoringEvent, 0, 3)
+	handlers := make([]EventHandler, 0, 3)
```

Comment:
nit: Maybe it's best not to pre-allocate the slices and rely on append to grow it. Tests will change in the future so this will give us one less thing to maintain.

Reply:
That's a really good point. Updated in  https://github.com/DataDog/datadog-agent/pull/12099/commits/3483bcc299ebda145257a11d10995abdb11148af .